### ISSUE TITLE
3.1, Prefix routing with the 'Route' class

### ISSUE THREAD

**Issue by davidyell** [cakephp/cakephp#8081]
**TL;DR**
Using a prefix route, without fallback routes and the `Route` class as the default route class, causes the passed controller param to be directly mapped to both `Controller::$name` and thus `Controller::$modelClass`.

**What's going on?**
I had an issue with the Crud plugin and lost my pagination. Turns out it was trying to match on the `Controller::$modelClass`, which was `packages` rather than `Packages`.

Chasing this all the way through it was being taken directly from the url. To fix it I changed my default route class to `InflectedRoute` and it was fine after that.

**Example application** 
https://github.com/davidyell/CakePHP3-Prefix-Routing/commit/56edff9169dbe4af00d7dada3a35c7489b71ff83

**Reproduction**
`example.com/examples` gives `Controller::$name === 'Examples'`
`example.com/foo/examples` gives `Controller::$name === 'examples'`

You can change the capitalization of the `Controller::$name` just by changing the url and refreshing.

**Comment by markstory** [comment]
I'll take a look and see if I can reproduce.

**Comment by markstory** [comment]
This is expected behavior. The default Route class does not apply any inflection. It's intended to be simple. The reason your unprefixed routes are inflecting is that your `/` scope uses `DashedRoute` when connecting the default routes.

Closing as the base Route class is working as it should and not being fancy :smile:

**Comment by ADmad** [comment]
@markstory I think you might have misunderstood the issue. The problem is casing of values of `Controller::$name` and `Controller::$modelClass` changes depending on route class used. That shouldn't happen. Shouldn't the values of those properties always be CamelCased?

**Comment by markstory** [comment]
@ADmad Yes I did misunderstand the issue.

**Comment by ADmad** [comment]
Closing a PR is opened.
